### ISSUE TITLE
Game hangs after pressing arrow keys with no beatmaps shown

### ISSUE THREAD

**Issue by smoogipoo** [ppy/osu#2069]
1. Go into song select.
2. Put in a search query that removes all beatmaps, like "aaaaaaaaaa".
3. Press any arrow key on your keyboard. You might have to press it twice if you follow these steps due to https://github.com/ppy/osu-framework/issues/1393.

**Comment by ProfessorLaw** [comment]
**tl;dr** : Index issue - Stuck because it will never reach an index when no beatmap can be found.

Code related thingies ahead because I need a technical advice.
The problem is in the ```SelectNext``` method from ```BeatmapCarousel```. It is stuck in a while loop.

In a normal case here's what we have :

```selectedBeatmap``` has a value
```originalIndex``` equals to the position of ```selectedBeatmap``` in the ```Items``` list (which basically contains every ```isPresent``` beatmap (& set) drawables)

However here's what we get when we have an empty result :

```selectedBeatmap``` is ```null```
```originalIndex``` equals to ```-1``` (because we have no reference to anything)

From there, we could simply tell to ```return``` if ```selectedBeatmap``` is ```null```, which is correct, but I think it's a bit more complicated.

The method already handles the case where "the beatmap is not there yet", and when ```Items``` list is empty. 
However, even with an empty resultset, ```Items``` is not empty.

The reason is that is seems to store drawables which have ```isPresent``` at true, however, some of them still return true even if they are not visible.
I think that we shouldn't loop on them as they are not visible anyway.

So I tried to filter the ```Items``` list to only keep visible drawables. This fixed the issue too.

This seems a lot for a small issue hehe.. But I prefer asking before making a pull request which might be wrong. What would you do?
